### PR TITLE
Sync: Allow detecting active plugins

### PR DESCRIPTION
**Pull request by withinboredom**
While working on something else, it looks like we aren't syncing whether a plugin is active, only that it's installed. This is a simple technique to allow us to get that information.

#### Changes proposed in this Pull Request:

* Adds a filter on the callable: `get_plugins` to report an `Active` state which will be `true` if the plugin is active, and `false` if not.
* Updated unit tests to pass with this new data

#### Testing instructions:

* Make sure unit tests pass
* Verify callable works correctly in multi-site and single-site installs.

cc: 
@ebinnion, @enejb since you touched this area of the code last
@eliorivero just because :p
@gravityrail since you asked ;)

### CODE REVIEW

File: sync/class.jetpack-sync-functions.php [hunk 1]
```diff
@@ -213,12 +213,27 @@ public static function normalize_www_in_url( $option, $url_function ) {
...
+
+		return $plugins;
+	}
+
 	public static function get_plugins() {
 		if ( ! function_exists( 'get_plugins' ) ) {
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
 		/** This filter is documented in wp-admin/includes/class-wp-plugins-list-table.php */
```
Comment (ebinnion) [inline sync/class.jetpack-sync-functions.php]: I believe this docblock should actually go above `apply_filters()`.

File: sync/class.jetpack-sync-functions.php [hunk 2]
```diff
@@ -213,12 +213,27 @@ public static function normalize_www_in_url( $option, $url_function ) {
...
 
+	public static function get_plugins_filter( $plugins ) {
+		$active_plugins = get_option( 'active_plugins' );
+
+		if ( is_multisite() ) {
+			$active_plugins = array_merge( get_option( 'active_sitewide_plugins' ) );
+		}
+
+		foreach ( $plugins as $plugin_file => &$plugin_meta ) {
+			$plugin_meta['Active'] = in_array( $plugin_file, $active_plugins );
```
Comment (ebinnion) [inline sync/class.jetpack-sync-functions.php]: Any reason why we're using an uppercase key instead of a lowercase one?

Comment (withinboredom) [inline sync/class.jetpack-sync-functions.php]: All the other keys are uppercase, I wanted to keep it consistent.

File: sync/class.jetpack-sync-functions.php (hunk 1 above)
Comment (withinboredom) [inline sync/class.jetpack-sync-functions.php]: 🤦‍♂️

**Comment by withinboredom** [comment]
Closing in favor of a better solution discussed in p1494277812297867-jpop-growth

**Pull request closed without merging**